<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Peter Ralph   University of Oregon   Dept of Data Science">
  <title>Simulation-based inference for ecology and evolution</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="./reveal.js/dist/reset.css">
  <link rel="stylesheet" href="./reveal.js/dist/reveal.css">
  <style>
    .reveal .sourceCode {  /* see #7635 */
      overflow: visible;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    /* The extra [class] is a hack that increases specificity enough to
       override a similar rule in reveal.js */
    ul.task-list[class]{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      font-size: inherit;
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="./reveal.js/dist/theme/simple.css" id="theme">
  <style type="text/css">

  .reveal { font-size: 24px; }

  .reveal h1 { font-size: 1.5em; } 

  .reveal h2 { font-size: 1.2em; } 

  .reveal .slides { text-align: left; }

  .reveal .slides figure { text-align: center; }

  .centered p { text-align: center; }
  img.centered { margin-left: auto; margin-right: auto; display: block; }

  /* apply this to the columns div */
  div.vcentered { display: flex; align-items: center; }

  div.bottomright { position: fixed; bottom: 0; right: 0; }

  .caption p { font-size: 75%; font-style: italic; color: #0006; clear: both; }

  .greyed { color: #0004; }

  img.floatright, div.floatright { float: right; text-align: right; }
  img.floatleft, div.floatleft { float: left; text-align: left; }

  img.centered { margin-left: auto; margin-right: auto; display: block; }

  .invisible { opacity: 0; }

  .reveal figcaption { display: none; }

  .reveal img { border: none; }

  .reveal section img { border: none; }

  .MathJax_Display { margin: 0; }

  img.finger { vertical-align: text-top; }

  .smaller { font-size: 50%; }
  .smallish { font-size: 80%; }

  .hide { opacity: 0; }

  .red { color: red; }
  .blue { color: blue; }

  .coverup { background: white; }

  div.column { overflow: visible; }

  </style>

  <div style="display: none">
  \[
  \usepackage{amsmath}
  \usepackage{amssymb}
  \usepackage{xcolor}

  \newcommand{\R}{\mathbb{R}}
  \newcommand{\Q}{\mathbb{Q}}
  \newcommand{\Z}{\mathbb{Z}}
  \newcommand{\N}{\mathbb{N}}
  \newcommand{\C}{\mathbb{C}}

  \renewcommand{\P}{\mathbb{P}}
  \newcommand{\E}{\mathbb{E}}

  \newcommand{\var}{\mathop{\mbox{Var}}}
  \newcommand{\cov}{\mathop{\mbox{cov}}}
  \newcommand{\median}{\mathop{\mbox{median}}}
  %\newcommand{\det}{\mathop{\mbox{det}}}
  \newcommand{\supp}{\mathop{\mbox{supp}}}
  \newcommand{\sgn}{\mathop{\mbox{sgn}}}

  \newcommand{\conv}{\mathop{\mbox{conv}}}
  \newcommand{\deq}{\stackrel{\scriptscriptstyle{d}}{=}}
  \newcommand{\dcv}{\stackrel{\scriptscriptstyle{d}}{\longrightarrow}}

  \]
  </div>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Simulation-based inference for ecology and
evolution</h1>
  <p class="author">Peter Ralph <br/> University of Oregon <br/> Dept of
Data Science</p>
  <p class="date">Quantitative and Computational Biology Seminar<br/>
University of Southern California // 4 December 2024</p>
</section>

<section class="slide level2">

<!--
title-slide-attributes:
    data-background-image: /path/to/title_image.png
    data-background-size: contain
-->
<!--
Abstract:
Modern datasets have made it possible to use genomic data to try to answer
increasingly detailed and practically relevant questions about evolution and
ecology. Although in many applications analytical inference methods are
infeasible or not available, simulation-based inference is possible for any
system that can be simulated. In this talk I will describe recent progress in
simulating large, evolving populations, and storing and analyzing the entire
genomic history of these simulations (using SLiM and the "tree sequence" data
structure). I will then give three examples of applications: understanding how
the balance of different types of natural selection, mutation rate variation,
and other processes produced the landscapes of genetic diversity in the great
apes; describing conditions under which coevolution can produce spatial
correlations in phenotypes such as seen in the classic Taricha newt-Thamnophis
garter snake system; and training a deep learning model to predict maps of
population density.
-->
<blockquote>
<p>UO is located on the traditional indigenous homeland of the Kalapuya
people. Kalapuya people were dispossessed of their indigenous homeland
by the United States government and forcibly removed. Today, Kalapuya
descendants are primarily citizens of the Confederated Tribes of Grand
Ronde and the Confederated Tribes of Siletz Indians, and continue to
make important contributions to their communities, to the UO, to Oregon,
and to the world.</p>
</blockquote>
<div class="fragment">
<div class="centered">
<p>USC: on unceded Tongva land</p>
</div>
</div>
</section>
<section id="outline-of-the-talk" class="slide level2">
<h2>Outline of the talk</h2>
<ol type="1">
<li>Genomes, and simulations</li>
<li>The tree sequence (but, mostly tomorrow)</li>
<li>Applications</li>
</ol>
<p><em>slides:</em> <a
href="https://petrelharp.github.io/usc-ceed-2024/usc-ceed-2024.slides.html">github.com/petrelharp/usc-ceed-2024</a></p>
</section>
<section id="motivating-problems" class="slide level2">
<h2>Motivating problems</h2>
<div class="columns">
<div class="column" style="width:60%;">
<ol type="1">
<li><p>What models are consistent with the observed spatial patterns of
coevolution between <em>Taricha</em> newts and <em>Thamnophis</em>
snakes?</p></li>
<li><p>How much natural selection is needed to explain the correlations
in genetic diversity along the genomes of the great apes?</p></li>
<li><p>How can we predict the landscape of population density using
spatial genomic data?</p></li>
</ol>
</div><div class="column" style="width:40%;">
<p><img data-src="figs/map_w_genomes.png" style="width:100.0%" /></p>
</div>
</div>
<div class="centered">
<p><img data-src="figs/murillo/greatapes_pi.png"
style="width:60.0%" /></p>
</div>
</section>
<section>
<section id="inference-with-genomes" class="title-slide slide level1">
<h1>Inference, with genomes</h1>

</section>
<section id="section" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_empty.png" /></p>
</section>
<section id="section-1" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_reality.png" /></p>
</section>
<section id="section-2" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_model_parameters.png" /></p>
</section>
<section id="section-3" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_model_parameters_inverse.png" /></p>
</section>
<section id="section-4" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_parameters_inverse_computer.png" /></p>
</section>
<section id="simulation-based-inference" class="slide level2">
<h2>Simulation-based inference</h2>
<div class="centered">
<p><img data-src="figs/modeling_parameters_inverse_computer.png"
style="width:60.0%" /></p>
</div>
<ul>
<li>bespoke confirmatory simulations</li>
<li>optimization of one or two parameters
<!-- *(if between-simulation noise is small)* --></li>
<li>Approximate Bayesian Computation (ABC)</li>
<li>deep learning</li>
</ul>
</section></section>
<section>
<section id="what-do-we-need" class="title-slide slide level1">
<h1>What do we need</h1>

</section>
<section id="section-5" class="slide level2">
<h2></h2>
<ol type="1">
<li><p>Fast simulation of genomes</p></li>
<li><p>Fast computation of summary statistics</p></li>
</ol>
</section>
<section id="wish-list" class="slide level2">
<h2>Wish list:</h2>
<div class="smallish">
<div class="columns">
<div class="column">
<p>Whole genomes, thousands of samples,<br />
from millions of individuals.</p>
<p><strong>Demography:</strong></p>
<ul>
<li>life history</li>
<li>separate sexes</li>
<li>selfing</li>
<li>polyploidy</li>
<li>species interactions</li>
</ul>
<p><strong>Geography:</strong></p>
<ul>
<li>discrete populations</li>
<li>continuous landscapes</li>
<li>barriers</li>
</ul>
<p><strong>History:</strong></p>
<ul>
<li>ancient samples</li>
<li>range shifts</li>
</ul>
</div><div class="column">
<p><strong>Natural selection:</strong></p>
<ul>
<li>selective sweeps</li>
<li>introgressing alleles</li>
<li>background selection</li>
<li>quantitative traits</li>
<li>incompatibilities</li>
<li>local adaptation</li>
</ul>
<p><strong>Genomes:</strong></p>
<ul>
<li>recombination rate variation</li>
<li>gene conversion</li>
<li>infinite-sites mutation</li>
<li>nucleotide models</li>
<li>context-dependence</li>
<li>mobile elements</li>
<li>inversions</li>
<li>copy number variation</li>
</ul>
</div>
</div>
</div>
</section>
<section id="enter-slim" class="slide level2">
<h2>Enter SLiM</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>by Ben Haller and Philipp Messer</p>
<p>an individual-based, scriptable forwards simulator</p>
<div class="caption">
<p><img data-src="figs/ben-haller.jpg" alt="Ben Haller" /> <em>Ben
Haller</em></p>
</div>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/slim-gui.png" alt="SLiM GUI" />
<figcaption aria-hidden="true">SLiM GUI</figcaption>
</figure>
<p><a href="https://messerlab.org/SLiM/">messerlab.org/SLiM</a></p>
</div>
</div>
</section>
<section id="section-6" class="slide level2">
<h2></h2>
<div class="smallish">
<div class="columns">
<div class="column">
<ul>
<li><s>Whole genomes,</s>*</li>
<li><s>thousands of samples, </s></li>
<li><s>from millions of individuals.</s>*</li>
</ul>
<p><strong>Demography:</strong></p>
<ul>
<li><s>life history</s></li>
<li><s>separate sexes</s>*</li>
<li><s>selfing</s></li>
<li>polyploidy*</li>
<li><s>species interactions</s></li>
</ul>
<p><strong>Geography:</strong></p>
<ul>
<li><s>discrete populations</s></li>
<li><s>continuous landscapes</s></li>
<li><s>barriers</s>*</li>
</ul>
<p><strong>History:</strong></p>
<ul>
<li><s>ancient samples</s></li>
<li><s>range shifts</s></li>
</ul>
</div><div class="column">
<p><strong>Natural selection:</strong></p>
<ul>
<li><s>selective sweeps</s></li>
<li><s>introgressing alleles</s></li>
<li><s>background selection</s></li>
<li><s>quantitative traits</s>*</li>
<li><s>incompatibilities</s>*</li>
<li><s>local adaptation</s>*</li>
</ul>
<p><strong>Genomes:</strong></p>
<ul>
<li><s>recombination rate variation</s></li>
<li><s>gene conversion</s></li>
<li><s>infinite-sites mutation</s></li>
<li><s>nucleotide models</s></li>
<li><s>context-dependence</s>*</li>
<li>mobile elements*</li>
<li><s>inversions</s></li>
<li>copy number variation</li>
</ul>
</div>
</div>
</div>
</section>
<section id="section-7" class="slide level2">
<h2></h2>
<ul>
<li><s>Whole genomes,</s>*</li>
</ul>
<!-- Tree sequences -->
</section></section>
<section>
<section id="the-tree-sequence" class="title-slide slide level1">
<h1>The tree sequence</h1>

</section>
<section id="meiosis-recombination" class="slide level2">
<h2>Meiosis &amp; Recombination</h2>
<div class="columns">
<div class="column" style="width:50%;">
<figure>
<img data-src="figs/recombination-cartoon-crop.png"
alt="recombination" />
<figcaption aria-hidden="true">recombination</figcaption>
</figure>
</div><div class="column" style="width:50%;">
<ul>
<li><p>You have two copies of each chromosome, one from each
parent.</p></li>
<li><p>When you make a gamete, the copies <em>recombine</em>.</p></li>
<li><p><em>Mutations</em> appear.</p></li>
</ul>
</div>
</div>
</section>
<section id="history-is-a-sequence-of-trees" class="slide level2">
<h2>History is a sequence of trees</h2>
<p>For a set of sampled chromosomes, at each position along the genome
there is a genealogical tree that says how they are related.</p>
<div class="fragment">
<figure>
<img data-src="figs/example.svg" alt="Trees along a chromosome" />
<figcaption aria-hidden="true">Trees along a chromosome</figcaption>
</figure>
</div>
</section>
<section class="slide level2">

<p>The <strong>succinct tree sequence</strong></p>
<div class="floatright">
<p>is a way to succinctly describe this, er, sequence of trees</p>
<p><em>and</em> the resulting genome sequences.</p>
<div class="caption">
<p><a
href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004842">Kelleher,
Etheridge, &amp; McVean</a></p>
</div>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<figure>
<img data-src="figs/tskit_logo.png" style="width:100.0%"
alt="tskit logo" />
<figcaption aria-hidden="true">tskit logo</figcaption>
</figure>
</div><div class="column" style="width:50%;">
<div class="floatright">
<figure>
<img data-src="figs/jerome.jpeg" style="width:80.0%"
alt="jerome kelleher" />
<figcaption aria-hidden="true">jerome kelleher</figcaption>
</figure>
<div class="caption">
<p>jerome kelleher</p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="file-sizes" class="slide level2">
<h2>File sizes</h2>
<div class="centered">
<figure>
<img data-src="figs/tsinfer_sizes.png" style="width:90.0%"
alt="file sizes" />
<figcaption aria-hidden="true">file sizes</figcaption>
</figure>
</div>
<div class="caption">
<p>100Mb chromosomes; from <a
href="https://www.nature.com/articles/s41588-019-0483-y">Kelleher et al
2018, <em>Inferring whole-genome histories in large population
datasets</em></a>, Nature Genetics</p>
</div>
<!-- Estimated sizes of files required to store the genetic variation data for a
simulated human-like chromosome (100 megabases) for up to 10 billion haploid
(5 billion diploid) samples. Simulations were run for 10 1 up to 10 7 haplotypes
using msprime [Kelleher et al., 2016], and the sizes of the resulting files plotted
(points). -->
</section></section>
<section>
<section id="application-to-genomic-simulations"
class="title-slide slide level1">
<h1>Application to genomic simulations</h1>

</section>
<section id="the-main-idea" class="slide level2">
<h2>The main idea</h2>
<p>Genomes are <em>big</em>. But a lot of the variants are neutral.</p>
<div class="fragment">
<p>If we <em>record the tree sequence</em> that relates everyone to
everyone else,</p>
<p>after the simulation is over we can put neutral mutations down on the
trees.</p>
</div>
<div class="fragment">
<p>Since neutral mutations don’t affect demography,</p>
<p>this is <em>equivalent</em> to having kept track of them
throughout.</p>
</div>
</section>
<section class="slide level2">

<p>This means recording the entire genetic history of
<strong>everyone</strong> in the population, <strong>ever</strong>.</p>
<div class="fragment">
<p>It is <em>not</em> clear this is a good idea.</p>
</div>
<div class="fragment">
<p>But, with a few tricks…</p>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<div class="caption">
<p>From Kelleher, Thornton, Ashander, and R. 2018, <a
href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006581">Efficient
pedigree recording for fast population genetics simulation</a>.</p>
<p>and Haller, Galloway, Kelleher, Messer, and R. 2018, <a
href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12968"><em>Tree‐sequence
recording in SLiM opens new horizons for forward‐time simulation of
whole genomes</em></a></p>
</div>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/jared.jpeg" style="width:35.0%"
alt="jared galloway" /> <img data-src="figs/jaime.jpg"
style="width:30.0%" alt="jaime ashander" /> <img
data-src="figs/ben-haller.jpg" style="width:30.0%"
alt="ben haller" /></p>
</div>
</div>
</div>
</div>
</section>
<section id="a-100x-speedup" class="slide level2">
<h2>A 100x speedup!</h2>
<div class="columns">
<div class="column" style="width:40%;">
<div class="centered">
<figure>
<img data-src="figs/slim_logo.png" style="width:100.0%"
alt="SLiM logo" />
<figcaption aria-hidden="true">SLiM logo</figcaption>
</figure>
</div>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p><img data-src="figs/slim_timing_no_msprime.png"
style="width:100.0%" /></p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<p><strong>For example:</strong></p>
<ul>
<li>genome as human chr7 (<span class="math inline">\(1.54 \times
10^8\)</span>bp)</li>
<li><span class="math inline">\(\approx\)</span> 10,000 diploids</li>
<li>500,000 overlapping generations</li>
<li>continuous, square habitat</li>
<li>selected mutations at rate <span
class="math inline">\(10^{-10}\)</span></li>
<li>neutral mutations added afterwards</li>
</ul>
<p><em>Runtime:</em> 8 hours on a single core</p>
<!-- Snewts -->
</section></section>
<section>
<section id="spatial-coevolution-snakes-and-newts"
class="title-slide slide level1">
<h1>Spatial coevolution: snakes and newts</h1>
<div class="columns">
<div class="column" style="width:70%;">
<figure>
<img data-src="figs/snewts/biowarfare_cartoon.gif"
alt="Battling newts and snakes" />
<figcaption aria-hidden="true">Battling newts and snakes</figcaption>
</figure>
</div><div class="column" style="width:30%;">
<div class="centered">
<figure>
<img data-src="figs/victoria.jpg" style="width:80.0%"
alt="Victoria Caudill" />
<figcaption aria-hidden="true">Victoria Caudill</figcaption>
</figure>
<p>Victoria Caudill</p>
<div class="caption greyed">
<p>Paper: ahead of print in <em>American Naturalist</em></p>
</div>
</div>
</div>
</div>
<div class="caption">
<p>Image from <a
href="https://evolution.berkeley.edu/biological-warfare-and-the-coevolutionary-arms-race/coevolution/">evolution.berkeley.edu</a></p>
</div>
</section>
<section id="section-8" class="slide level2"
data-background-image="figs/snewts/snewt-lit.png"
data-background-position="top" data-background-width="60%">
<h2 data-background-image="figs/snewts/snewt-lit.png"
data-background-position="top" data-background-width="60%"></h2>
<div class="caption bottomright smallish">
<p>newt: <a
href="https://en.m.wikipedia.org/wiki/File:California_Newt_%28Taricha_torosa%29_%284415749606%29.jpg">CC
by Steve Jurvetson</a></p>
<p>snake: <a
href="https://commons.wikimedia.org/wiki/File:California_Red-Sided_Garter_Snake_(Thamnophis_sirtalis_infernalis).jpg">CC
by Jaden Clark</a></p>
</div>
</section>
<section class="slide level2">

<figure>
<img data-src="figs/snewts/reimche2020-fig1.jpg" style="width:70.0%"
alt="Fig 1 from Reimche et al 2020" />
<figcaption aria-hidden="true">Fig 1 from Reimche et al
2020</figcaption>
</figure>
<div class="caption">
<p><a
href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/1365-2656.13212">Fig
1, Reimche et al 2020</a></p>
</div>
</section>
<section class="slide level2">

<figure>
<img data-src="figs/snewts/hanifin2008-fig2.png" style="width:70.0%"
alt="Fig 2 from Hanifin et al 2008" />
<figcaption aria-hidden="true">Fig 2 from Hanifin et al
2008</figcaption>
</figure>
<div class="caption">
<p><a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.0060060">Fig
2, Hanifin et al 2008</a></p>
</div>
</section>
<section id="why-and-how" class="slide level2">
<h2>Why, and how?</h2>
<p><img data-src="figs/snewts/brodie2002abstract.png"
style="width:100.0%" /></p>
</section>
<section id="a-spatial-co-evolutionary-simulation" class="slide level2">
<h2>A spatial co-evolutionary simulation</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li><p>continuous space</p></li>
<li><p>local density-dependent mortality</p></li>
<li><p>additive, costly traits (“toxicity” and “resistance”)</p></li>
<li><p>various genetic architectures</p></li>
<li><p>snakes may encounter nearby newts, outcome depends on difference
in traits:</p>
<ul>
<li>snake eats newt, gets fitness benefit, or</li>
<li>snake dies, newt escapes</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/snewts/slim-screenshot.png"
alt="Screenshot of a SLiM simulation" />
<figcaption aria-hidden="true">Screenshot of a SLiM
simulation</figcaption>
</figure>
</div>
</div>
</section>
<section id="no-spatial-correlation-on-a-flat-map" class="slide level2">
<h2>No spatial correlation on a flat map</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="figs/snewts/nocor_map.png" /></p>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/snewts/nocor_boxplots.png" /></p>
</div>
</div>
</section>
<section id="add-some-heterogeneity" class="slide level2">
<h2>Add some heterogeneity</h2>
<p><img data-src="figs/snewts/gradient-maps.png" /></p>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="figs/snewts/cor_map.png" /></p>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/snewts/cor_boxplots.png" /></p>
</div>
</div>
</section>
<section id="conclusions" class="slide level2">
<h2>Conclusions</h2>
<div class="columns">
<div class="column" style="width:50%;">
<figure>
<img data-src="figs/snewts/hanifin2008-fig2.png"
alt="Fig 2 from Hanifin et al 2008" />
<figcaption aria-hidden="true">Fig 2 from Hanifin et al
2008</figcaption>
</figure>
</div><div class="column" style="width:50%;">
<ul>
<li><p>such large correlated differences across the landscape unlikely
to be due to nonadaptive forces</p></li>
<li><p>spatial heterogeneity in ecological factors much more
plausible</p></li>
<li><p>trait genetic architecture has little effect, given sufficient
variation</p></li>
</ul>
<div class="caption floatright">
<figure>
<img data-src="figs/victoria.jpg" class="floatright" style="width:20.0%"
alt="Victoria Caudill" />
<figcaption aria-hidden="true">Victoria Caudill</figcaption>
</figure>
<p>Victoria Caudill</p>
</div>
</div>
</div>
<!-- Great apes -->
</section></section>
<section>
<section id="landscapes-of-genetic-diversity"
class="title-slide slide level1">
<h1>Landscapes of genetic diversity</h1>
<p><a
href="https://academic.oup.com/genetics/advance-article/doi/10.1093/genetics/iyae006/7577595"><img
data-src="figs/murillo/genetics.png" class="floatleft"
style="width:70.0%"
alt="https://academic.oup.com/genetics/advance-article/doi/10.1093/genetics/iyae006/7577595" /></a></p>
<div class="caption floatright">
<p><img data-src="figs/murillo.jpeg" style="width:80.0%" /></p>
<p><a href="https://www.m-rodrigues.me/">Murillo Rodrigues</a></p>
</div>
</section>
<section id="landscapes-of-genetic-diversity-within-groups-of-species"
class="slide level2">
<h2>Landscapes of genetic diversity within groups of species</h2>
<p><img data-src="figs/murillo/correlated-pi-examples.png"
style="width:80.0%" /></p>
<div class="caption floatright">
<p><a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000391">Stankowski,
Chase et al. (2019)</a>; <a
href="https://onlinelibrary.wiley.com/doi/10.1111/mec.14083">Van Doren
et al. (2017)</a>; <a
href="https://academic.oup.com/evlett/article/1/3/118/6697077">Buri
(2017)</a></p>
</div>
</section>
<section id="the-study-system-us" class="slide level2">
<h2>The study system: “us”</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>High quality genomic data for 5 species</li>
<li>Deep divergence, spanning ~60N generations</li>
<li>Conservation of genes, recombination rate and other genomic
features</li>
</ul>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/murillo/greatapes.png"
alt="Phylogeny of the great apes from Javier-Prado et al. (2013) Nature" />
<figcaption aria-hidden="true">Phylogeny of the great apes from
Javier-Prado et al. (2013) Nature</figcaption>
</figure>
<div class="caption flushright">
<p><a href="https://www.nature.com/articles/nature12228">Javier-Prado et
al. (2013)</a></p>
</div>
</div>
</div>
</section>
<section id="genetic-diversity-in-the-great-apes-chromosome-12"
class="slide level2">
<h2>Genetic diversity in the great apes: chromosome 12</h2>
<p><img data-src="figs/murillo/greatapes_pi.png" /></p>
</section>
<section id="genetic-divergence-between-the-great-apes-chromosome-12"
class="slide level2">
<h2>Genetic divergence between the great apes: chromosome 12</h2>
<p><img data-src="figs/murillo/greatapes_dxy.png" /></p>
</section>
<section id="goals" class="slide level2"
data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">
<h2 data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">Goals</h2>
<ul>
<li>How correlated are landscapes for closely related species?</li>
<li>How much is due to shared footprints of
<ul>
<li>history?</li>
<li>selection? what kind?</li>
<li>mutational processes?</li>
</ul></li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="invisible" style="height:80.0%" /></p>
</section>
<section id="slimulations" class="slide level2">
<h2>SLiMulations</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Chromosome-scale simulation (chr12)</li>
<li>selection on annotated exons (from Ensembl via <a
href="https://popsim-consortium.github.io/stdpopsim-docs/main/catalog.html#sec_catalog_HomSap_annotations">stdpopsim</a>)</li>
<li>deCode genetic map</li>
<li>branches simulated independently with SLiM and merged with <a
href="https://tskit.dev/pyslim/docs/latest/vignette_parallel_phylo.html">tskit</a></li>
<li>runtime: hours to weeks</li>
<li>total simulations: 56</li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/murillo/demog_tubes_greatapes.png" /></p>
</div>
</div>
</section>
<section id="back-to-the-data" class="slide level2">
<h2>Back to the data</h2>
<div class="centered">
<p><img data-src="figs/murillo/greatapes_pi.png" style="width:70.0%" />
<img data-src="figs/murillo/greatapes_dxy.png"
style="width:70.0%" /></p>
</div>
</section>
<section id="section-9" class="slide level2">
<h2></h2>
<p><img
data-src="figs/murillo/scatter-examples-pidxy_win-size_1000000_merged-mask_True_state_all_curr_all_prop-acc_0.4.png" /></p>
<p><img data-src="figs/murillo/trees-with-dt.png" /></p>
</section>
<section
id="the-data-empirical-correlations-for-chromosome-12-in-the-great-apes"
class="slide level2">
<h2>The data: empirical correlations for chromosome 12 in the great
apes</h2>
<p><img data-src="figs/murillo/cor-pidxy-dT_data.png" /></p>
</section>
<section id="no-correlation-under-neutrality" class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">No
correlation under neutrality</h2>
<div class="columns">
<div class="column" style="width:30%;">
<ul>
<li>correlations decay to zero quickly with split time</li>
</ul>
</div><div class="column" style="width:70%;">
<p><img data-src="figs/murillo/corr-pidxy-land-panel_neutral.png" /></p>
</div>
</div>
</section>
<section id="mutation-rate-variation-can-contribute"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Mutation rate variation can contribute</h2>
<div class="columns">
<div class="column" style="width:30%;">
<ul>
<li>but correlations do not decay with time</li>
</ul>
</div><div class="column" style="width:70%;">
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-mutvar-sims_1.png"
class="centered" style="width:80.0%" /></p>
</div>
</div>
</section>
<section id="negative-selection-produces-weak-correlations"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Negative selection produces weak
correlations</h2>
<p><img
data-src="figs/murillo/cor-pidxy-dT_selected-negative.png" /></p>
</section>
<section id="positive-selection-produces-strong-correlations"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Positive selection produces strong
correlations</h2>
<ul>
<li>sometimes too strong</li>
</ul>
<div class="columns">
<div class="column" style="width:30%;">

</div><div class="column" style="width:70%;">
<p><span class="math display">\[ \vphantom{ d_{xy} = \pi_\text{anc}
\nearrow + \mu T_\text{MRCA} \searrow } \]</span></p>
</div>
</div>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="centered" /></p>
</section>
<section id="the-best-fit-both" class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">The best fit: both!</h2>
<ul>
<li>chosen from 37 distinct simulations</li>
<li>deleterious mutations: <span class="math inline">\(1.2 \times
10^{-8}\)</span></li>
<li>beneficial mutations: <span
class="math inline">\(10^{-12}\)</span></li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-both.png"
class="centered" style="width:60.0%" /></p>
</section>
<section id="conclusions-1" class="slide level2"
data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">
<h2 data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom"
data-background-size="60%">Conclusions</h2>
<ul>
<li>positive selection most necessary for a good fit</li>
<li>best guess: <span class="math inline">\(\approx 10\%\)</span> of
fixations on human lineage due to positive selection (1/250 years);</li>
<li><span class="math inline">\(\approx\)</span> 70% of mutations in
exons deleterious</li>
<li>fixation rate in exons reduced by half</li>
<li>GC-biased gene conversion causes “smile” at ends of chromosomes</li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="invisible" style="height:80.0%" /></p>
</section></section>
<section>
<section id="maps-of-population-density"
class="title-slide slide level1">
<h1>Maps of population density</h1>
<div class="centered">
<p><a href="https://doi.org/10.1111/1755-0998.14005"><img
data-src="figs/mapNN/paper.png" style="width:70.0%" /></a></p>
</div>
<div class="caption floatright">
<p><img data-src="figs/mapNN/chris.jpeg" style="width:23.0%" /> <img
data-src="figs/gilia.jpeg" style="width:20.0%" /> <img
data-src="figs/andy_22.jpg" style="width:15.0%" /></p>
<p>with <a
href="https://scholar.google.com/citations?user=ZHHGVn8AAAAJ&amp;hl=en">Chris
C R Smith</a> Gilia Patterson &amp; <a
href="https://kr-colab.github.io">Andy Kern</a>,</p>
</div>
</section>
<section id="goal" class="slide level2">
<h2>Goal</h2>
<p><em>This</em> but with parameters <span
class="math inline">\(\mu\)</span>, <span
class="math inline">\(\theta\)</span> the <strong>maps</strong> of mean
density (<span class="math inline">\(K\)</span>) and dispersal distance
(<span class="math inline">\(\sigma\)</span>):</p>
<p><img data-src="figs/modeling_parameters_inverse_computer.png" /></p>
</section>
<section id="design" class="slide level2">
<h2>Design</h2>
<p><img data-src="figs/mapNN/architecture.png" /></p>
</section>
<section id="training" class="slide level2">
<h2>Training</h2>
<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>randomly generated landscapes of <span
class="math inline">\(K\)</span> and <span
class="math inline">\(\sigma\)</span>,
<ul>
<li>passed to spatial SLiMulations</li>
<li>and used as the target</li>
</ul></li>
<li>data: <span class="math inline">\(\sim 100\)</span> sampled <span
class="math inline">\(10^8\)</span>bp genomes</li>
<li>450 random sample pairs</li>
</ul>
</div><div class="column" style="width:40%;">
<p><img data-src="figs/computer.png" style="width:80.0%" /></p>
</div>
</div>
</section>
<section id="results" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/feems_1a.png" /></p>
</section>
<section id="results-1" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/feems_1.png" /></p>
</section>
<section id="results-2" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/methods_comparison.png" /></p>
</section>
<section id="results-3" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/methods_comparison.png" /></p>
</section>
<section id="where-next" class="slide level2">
<h2>Where next?</h2>
<p>Most of the <em>power</em> and <em>robustness</em> comes from a good
ensemble of simulations.</p>
<div class="fragment">
<p>We need more examples!</p>
</div>
</section></section>
<section>
<section id="wrap-up" class="title-slide slide level1">
<h1>Wrap-up</h1>

</section>
<section id="software-development-goals" class="slide level2">
<h2>Software development goals</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>open</li>
<li>welcoming and supportive</li>
<li>reproducible and well-tested</li>
<li>backwards compatible</li>
<li>well-documented</li>
<li>capacity building</li>
</ul>
<div class="centered">
<figure>
<img data-src="figs/popsim.png" style="width:35.0%" alt="popsim logo" />
<figcaption aria-hidden="true">popsim logo</figcaption>
</figure>
</div>
<p><a
href="https://popsim-consortium.github.io/stdpopsim-docs/stable/index.html">PopSim
Consortium</a></p>
</div><div class="column" style="width:50%;">
<div class="centered">
<figure>
<img data-src="figs/tskit_logo.png" style="width:60.0%"
alt="tskit logo" />
<figcaption aria-hidden="true">tskit logo</figcaption>
</figure>
<p><a href="https://tskit.dev">tskit.dev</a></p>
<figure>
<img data-src="figs/slim_logo.png" style="width:80.0%"
alt="SLiM logo" />
<figcaption aria-hidden="true">SLiM logo</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="thanks" class="slide level2">
<h2>Thanks!</h2>
<div class="columns">
<div class="column" style="width:40%;">
<div style="font-size: 85%;">
<ul>
<li>Andy Kern</li>
<li>Nate Pope</li>
<li>Victoria Caudill</li>
<li>Murillo Rodrigues</li>
<li>Gilia Patterson</li>
<li>Chris Smith</li>
<li>Thomas Forest</li>
<li>Jiseon Min</li>
<li>Clara Rehmann</li>
<li>Anastasia Teterina</li>
<li>Angel Rivera-Colon <!--
- Bruce Edelman
- Matt Lukac
- Saurabh Belsare
- Gabby Coffing
- Jeff Adrion
- CJ Battey
- Jared Galloway
--></li>
<li>the rest of <a href="https://kr-colab.github.io/people">the
Co-Lab</a></li>
</ul>
<p>Funding:</p>
<ul>
<li>NIH NIGMS</li>
<li>NSF DBI</li>
</ul>
</div>
<p><img src="figs/KernRalph_5x5.png" alt="KR-colab logo" style="width: 50%; margin: 0px;"/></p>
</div><div class="column" style="width:60%;">
<div class="columns">
<div class="column" style="width:30%;">
<div style="font-size: 85%;">
<ul>
<li>Jerome Kelleher</li>
<li>Ben Haller</li>
<li>Yan Wong</li>
<li>Ben Jeffery</li>
<li>Sam Tallman</li>
<li>Duncan Mbuli-Robertson</li>
<li>Hanbin Lee</li>
<li>Gregor Gorjanc</li>
<li>Elsie Chevy</li>
<li>Madeline Chase</li>
<li>Sean Stankowski</li>
<li>Matt Streisfeld <!--
- Georgia Tsambos
- Jaime Ashander
- Jared Galloway
- Gideon Bradburd
- Bill Cresko
- Alison Etheridge
- Evan McCartney-Melstad
- Brad Shaffer
--></li>
</ul>
</div>
</div><div class="column" style="width:30%;">
<div class="centered" style="margin-top: -60px;">
<p><img src="figs/tskit_logo.png" alt="tskit logo" style="width: 60%; margin: 10px;"/>
<img src="figs/slim_logo.png" alt="SLiM logo" style="width: 90%; margin: 10px;"/></p>
</div>
</div>
</div>
<div style="margin-top: -30px;">
<p><img data-src="figs/colab.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section>
<section id="section-10" class="slide level2"
data-background-image="figs/guillemots_thanks.png"
data-background-position="center" data-background-size="50%">
<h2 data-background-image="figs/guillemots_thanks.png"
data-background-position="center" data-background-size="50%"></h2>
</section></section>
    </div>
  </div>

  <script src="./reveal.js/dist/reveal.js"></script>

  <!-- reveal.js plugins -->
  <script src="./reveal.js/plugin/notes/notes.js"></script>
  <script src="./reveal.js/plugin/search/search.js"></script>
  <script src="./reveal.js/plugin/zoom/zoom.js"></script>
  <script src="./reveal.js/plugin/math/math.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
        // Display controls in the bottom right corner
        controls: true,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: true,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'bottom-right',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: true,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: false,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: true,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'default',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: true,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'fade',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        math: {
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [
          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    </body>
</html>
